### PR TITLE
doc(pubsub): create page for configuration options

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10198)

### CODE REVIEW

File: google/cloud/pubsub/options.h [hunk 1]
```diff
@@ -50,11 +50,13 @@ namespace pubsub {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 
 /// The retry policy
+/// @ingroup pubsub-options
```
Comment (dbolduc) [inline google/cloud/pubsub/options.h:53]: optional nit: use `/**` ?

File: google/cloud/pubsub/options.h [hunk 2]
```diff
@@ -50,11 +50,13 @@ namespace pubsub {
...
 struct RetryPolicyOption {
   using Type = std::shared_ptr<pubsub::RetryPolicy>;
 };
 
 /// The backoff policy
+/// @ingroup pubsub-options
```
Comment (dbolduc) [inline google/cloud/pubsub/options.h:59]: same optional nit

**Review by dbolduc — APPROVED** [review]

File: google/cloud/pubsub/options.h (hunk 1 above)
Comment (coryan) [reply google/cloud/pubsub/options.h]: Done.

File: google/cloud/pubsub/options.h (hunk 2 above)
Comment (coryan) [reply google/cloud/pubsub/options.h]: Thanks

**Pull request merged**